### PR TITLE
Methods table and overview update

### PR DESCRIPTION
**Pull request by growlix**
This PR replaces the Methods Overview with a table, because they seem redundant and it doesn't make sense to maintain both.

### CODE REVIEW

**Comment by dblalock** [comment]
Am I correct that it's just [docs/source/tables/methods_table.md](https://github.com/mosaicml/composer/pull/1436/files#diff-aa663781616b8488877b6dd7937fa60e1a6af7f8f76e31fa679dbb9184049529), [docs/source/tables/algorithms_table.md](https://github.com/mosaicml/composer/pull/1436/files#diff-d8339b74db55b1b96b6d30a12a412058808ce92b36d035b72806b7415572effd), and [docs/source/index.rst](https://github.com/mosaicml/composer/pull/1436/files#diff-4eec6cec5f6fab1548b85433ea8ca81315ae165db4b7f84019f287df9015699f) that have changed, and the rest of the changes are from branching off of your method cards branch?

File: docs/source/tables/methods_table.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+[EMA](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/ema)|`cf.compute_ema`|[Many](https://www.mosaicml.com)|Exponentially weighted moving average of model parameters during training|❌
+[Factorize](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/factorize)|`cf.apply_factorization`|[Many](https://www.mosaicml.com)|Factorize GEMMs into smaller GEMMs|❌
+[Fused LayerNorm](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/fused_layernorm)|`cf.apply_fused_layernorm`|[Many](https://www.mosaicml.com)|Replaces `torch.nn.LayerNorm` with `apex.normalization.fused_layer_norm`|✅
+[Gated Linear Units](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/gated_linear_units)|`cf.apply_gated_linear_units`|[Dauphin et al., 2016](https://arxiv.org/abs/1612.08083)|Adds gating operation to the input of feed-forward block.|❌
+[GhostBatchNorm](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/ghost_batchnorm)|`cf.apply_ghost_batchnorm`|[Dimitriou et al, 2020](https://arxiv.org/abs/2007.08554)|Use smaller samples to compute batchnorm|❌
```
Comment (dblalock) [inline docs/source/tables/methods_table.md:17]:
```suggestion
[GhostBatchNorm](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/ghost_batchnorm)|`cf.apply_ghost_batchnorm`|[Dimitriou et al, 2020](https://arxiv.org/abs/2007.08554)|Normalizes as if the batch size were smaller|❌
```

File: docs/source/tables/methods_table.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# 🃏 Methods Table
+
+Name|Functional|Attribution|tl;dr|Math-Equivalent
+----|----------|-----------|-----|----
+[Alibi](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/alibi)|`cf.apply_alibi`|[Press et al, 2021](https://arxiv.org/abs/2108.12409v1)|Replace attention with AliBi|❌
```
Comment (dblalock) [inline docs/source/tables/methods_table.md:5]:
```suggestion
[Alibi](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/alibi)|`cf.apply_alibi`|[Press et al, 2021](https://arxiv.org/abs/2108.12409v1)|Adds structured biases to attention matrices|❌
```

File: docs/source/tables/methods_table.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+[AugMix](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/augmix)|`cf.augmix_image`|[Hendrycks et al, 2020](http://arxiv.org/abs/1912.02781)|Image-perserving data augmentations|❌
```
Comment (dblalock) [inline docs/source/tables/methods_table.md:6]:
```suggestion
[AugMix](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/augmix)|`cf.augmix_image`|[Hendrycks et al, 2020](http://arxiv.org/abs/1912.02781)|Adds convex combinations of data augmentations|❌
```

File: docs/source/tables/methods_table.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+[BlurPool](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/blurpool)|`cf.apply_blurpool`|[Zhang, 2019](https://arxiv.org/abs/1904.11486)|applies blur before pooling|❌
```
Comment (dblalock) [inline docs/source/tables/methods_table.md:7]:
```suggestion
[BlurPool](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/blurpool)|`cf.apply_blurpool`|[Zhang, 2019](https://arxiv.org/abs/1904.11486)|Adds anti-aliasing before downsampling operations|❌
```

**Review by dblalock — COMMENTED** [review]
Main thing is being consistent about "Verb the noun" vs "Verbs the noun" in descriptions.

File: composer/algorithms/channels_last/README.md [hunk 5]
```diff
@@ -88,10 +88,19 @@ If the model weights are instead initialized in NHWC format, PyTorch will automa
 
 We currently implement this method by casting the user’s model to channels-last format (no changes to the dataloader are necessary). When the first convolution operation receives its input activation, it will automatically convert it to NHWC format, after which the memory format will persist for the remainder of the network (or until it reaches a layer that cannot support having channels last).
 
+> ✅ Channels Last Improves Training Speed
+>
+> In our experiments, Channels Last improves the attainable tradeoffs between training speed and the final quality of the trained model.
+> We recommend Channels Last for training convolutional networks.
```
Comment (moinnadeem) [inline composer/algorithms/channels_last/README.md:94]: This isn't universally true, right? Doesn't it slow down UNet? Should we add a less strong statement here?

**Review by moinnadeem — APPROVED** [review]
LGTM, small caveat on Channels Last

**Pull request closed without merging**